libs/extended/src/System/Logger/Extended.hs

```diff
@@ -1,32 +1,90 @@
+{-# LANGUAGE DeriveAnyClass #-}
 {-# LANGUAGE GeneralizedNewtypeDeriving #-}
-{-# LANGUAGE DerivingStrategies         #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
 -- | Tinylog convenience things.
 module System.Logger.Extended
     ( module Log
+    , LogFormat(..)
     , mkLogger
     , mkLogger'
     , LoggerT(..)
     , runWithLogger
+    , netStringsToLogFormat
     ) where
 
-import Imports
 import Control.Monad.Catch
+import Data.Aeson
+import Data.Aeson.Encoding (pair, list, text)
 import Database.CQL.IO
+import GHC.Generics
+import Imports
 import System.Logger as Log
-
-import qualified Data.ByteString.Lazy.Char8 as L
+import Data.String.Conversions (cs)
 import qualified Data.ByteString.Lazy.Builder as B
+import qualified Data.ByteString.Lazy.Char8 as L
 import qualified System.Logger.Class as LC
 
--- TODO(arianvp): Get rid of boolean blindness
--- TODO(arianvp): Add JSON log format. This will make our lives a lot easier
---    This will add a dependency on aeson for this package,
---    but it already transitively depended on it through imports.
---    Interestingly, the only place where imports uses Aeson
---    is in the Orphans module which defines Aeson Orphans instances for Log.Level.
---    So while we're at it, we should probably move those orphans here.
-mkLogger :: Log.Level -> Bool -> IO Log.Logger
-mkLogger lvl netstr = Log.new
+
+deriving instance Generic LC.Level
+instance FromJSON LC.Level
+instance ToJSON LC.Level
```

Comment:
i would move these to `System.Logger.Extended.Orphans`, or `System.Logger.Orphans`.  (This way we won't accidentally add more orphans because there is no warning about it.)  [ignore if you disagree]